apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs

```diff
@@ -14,256 +14,77 @@
-
-        [Fact]
-        public Task BindInt64EmptyArray() => TestBindNonNull(
-            SpannerDbType.ArrayOf(SpannerDbType.Int64),
-            new long[] { });
-
-        [Fact]
-        public async Task BindNumeric()
-        {
-            await TestBindNonNull(SpannerDbType.Numeric, SpannerNumeric.Parse("1.0"), r => r.GetNumeric(0));
-        }
-
-        [Fact]
-        public async Task BindNumericArray()
-        {
-            await TestBindNonNull(
-                SpannerDbType.ArrayOf(SpannerDbType.Numeric),
-                new SpannerNumeric?[] {SpannerNumeric.Parse("0.0"), null, SpannerNumeric.Parse("1.0")});
-        }
-
-        [Fact]
-        public async Task BindNumericEmptyArray()
-        {
-            await TestBindNonNull(SpannerDbType.ArrayOf(SpannerDbType.Numeric), new SpannerNumeric[] { });
-        }
-
-        [Fact]
-        public Task BindString() => TestBindNonNull(SpannerDbType.String, "abc", r => r.GetString(0));
-
-        [Fact]
-        public Task BindStringArray() => TestBindNonNull(
-            SpannerDbType.ArrayOf(SpannerDbType.String),
-            new[] {"abc", null, "123"});
-
-        [Fact]
-        public Task BindStringEmptyArray() => TestBindNonNull(
-            SpannerDbType.ArrayOf(SpannerDbType.String),
-            new string[] { });
-
-        [Fact]
-        public Task BindTimestamp() => TestBindNonNull(
-            SpannerDbType.Timestamp,
-            new DateTime(2017, 5, 26, 15, 0, 0));
-
-        [Fact]
-        public Task BindTimestampArray() => TestBindNonNull(
-            SpannerDbType.ArrayOf(SpannerDbType.Timestamp),
-            new DateTime?[]
-                {new DateTime(2017, 5, 26, 3, 15, 0), null, new DateTime(2017, 5, 9, 12, 30, 0)});
-
-        [Fact]
-        public Task BindTimestampEmptyArray() => TestBindNonNull(
-            SpannerDbType.ArrayOf(SpannerDbType.Timestamp),
-            new DateTime?[] { });
-
-        [Fact]
-        [Trait(Constants.SupportedOnEmulator, Constants.No)]
-        public async Task BindJson()
-        {
-            // The emulator doesn't yet support the JSON type.
-            await TestBindNonNull(SpannerDbType.Json, "{\"key\":\"value\"}", r => r.GetString(0));
-        }
-
-        [Fact]
-        [Trait(Constants.SupportedOnEmulator, Constants.No)]
-        public async Task BindJsonArray()
-        {
-            // The emulator doesn't yet support the JSON type.
-            await TestBindNonNull(
-                SpannerDbType.ArrayOf(SpannerDbType.Json),
-                new string[] { "{\"key\":\"value\"}", null, "{\"other-key\":\"other-value\"}" });
-        }
-
-        [Fact]
-        [Trait(Constants.SupportedOnEmulator, Constants.No)]
-        public async Task BindJsonEmptyArray()
-        {
-            // The emulator doesn't yet support the JSON type.
-            await TestBindNonNull(SpannerDbType.ArrayOf(SpannerDbType.Json), new string[] { });
-        }
+    [SkippableTheory]
```

Comment:
Is the trait not being used here because this is a theory? So we need to use skip instead?

Would it make sense to separate the test in two theories, one that is always supported and one that is not supported for the emulator so we can use the trait always? Are there other similar tests out there that we could apply the same rationale to?